app/scripts/metamask-controller.js

```diff
@@ -1168,6 +1185,148 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  // eth_decryptMessage methods
+
+  /**
+  * Called when a dapp uses the eth_decryptMessage method.
+  *
+  * @param {Object} msgParams - The params of the message to sign & return to the Dapp.
+  * @param {Function} cb - The callback function called with the signature.
+  * Passed back to the requesting Dapp.
+  */
+  async newRequestDecryptMessage (msgParams, req) {
+    const promise = this.decryptMessageManager.addUnapprovedMessageAsync(msgParams, req)
+    this.sendUpdate()
+    this.opts.showUnconfirmedMessage()
+    return promise
+  }
+
+  /**
+  * Only decypt message and don't touch transaction state
+  *
+  * @param {Object} msgParams - The params of the message to decrypt & return to the Dapp.
```

Comment:
```suggestion
  * @param {Object} msgParams - The params of the message to decrypt
```
I'd suggest removing that last part of the description - not that it's inaccurate technically, but it implied to me when I first read it that this function returned the decrypted message to the dapp.